yellowstone-grpc-tools/src/scylladb/types.rs

```diff
@@ -0,0 +1,705 @@
+            0 => Ok(BlockchainEventType::AccountUpdate),
+            1 => Ok(BlockchainEventType::NewTransaction),
+            x => Err(anyhow!("Unknown LogEntryType equivalent for {:?}", x)),
+        }
+    }
+}
+
+impl From<BlockchainEventType> for i16 {
+    fn from(val: BlockchainEventType) -> Self {
+        match val {
+            BlockchainEventType::AccountUpdate => 0,
+            BlockchainEventType::NewTransaction => 1,
+        }
+    }
+}
+
+impl SerializeCql for BlockchainEventType {
+    fn serialize<'b>(
+        &self,
+        typ: &scylla::frame::response::result::ColumnType,
+        writer: scylla::serialize::CellWriter<'b>,
+    ) -> Result<
+        scylla::serialize::writers::WrittenCellProof<'b>,
+        scylla::serialize::SerializationError,
+    > {
+        let x: i16 = (*self).into();
+        SerializeCql::serialize(&x, typ, writer)
+    }
+}
+
+impl FromCqlVal<CqlValue> for BlockchainEventType {
+    fn from_cql(cql_val: CqlValue) -> Result<Self, scylla::cql_to_rust::FromCqlValError> {
+        match cql_val {
+            CqlValue::SmallInt(x) => x.try_into().map_err(|_| FromCqlValError::BadVal),
+            _ => Err(FromCqlValError::BadCqlType),
+        }
+    }
+}
+
+#[derive(SerializeRow, Clone, Debug, FromRow, DeepSizeOf)]
+pub struct BlockchainEvent {
+    // Common
+    pub shard_id: ShardId,
+    pub period: ShardPeriod,
+    pub producer_id: ProducerId,
+    pub offset: ShardOffset,
+    pub slot: i64,
+    pub entry_type: BlockchainEventType,
+
+    // AccountUpdate
+    pub pubkey: Pubkey,
+    pub lamports: i64,
+    pub owner: Pubkey,
+    pub executable: bool,
+    pub rent_epoch: i64,
+    pub write_version: i64,
+    pub data: Vec<u8>,
+    pub txn_signature: Option<Vec<u8>>,
+
+    // Transaction
+    pub signature: Vec<u8>,
+    pub signatures: Vec<Vec<u8>>,
+    pub num_required_signatures: i32,
+    pub num_readonly_signed_accounts: i32,
+    pub num_readonly_unsigned_accounts: i32,
+    pub account_keys: Vec<Vec<u8>>,
+    pub recent_blockhash: Vec<u8>,
+    pub instructions: Vec<CompiledInstr>,
+    pub versioned: bool,
+    pub address_table_lookups: Vec<MessageAddrTableLookup>,
+    pub meta: TransactionMeta,
+}
+
+type Pubkey = [u8; 32];
+
+#[derive(SerializeRow, Clone, Debug, DeepSizeOf)]
+pub struct AccountUpdate {
+    pub slot: i64,
+    pub pubkey: Pubkey,
+    pub lamports: i64,
+    pub owner: Pubkey,
+    pub executable: bool,
+    pub rent_epoch: i64,
+    pub write_version: i64,
+    pub data: Vec<u8>,
+    pub txn_signature: Option<Vec<u8>>,
+}
+
+fn try_vec_into<U: fmt::Debug, I: IntoIterator>(
+    it: I,
+) -> Result<Vec<U>, <I::Item as TryInto<U>>::Error>
+where
+    I::Item: TryInto<U> + fmt::Debug,
+    <I::Item as TryInto<U>>::Error: fmt::Debug,
+{
+    let mut res = Vec::new();
+    for x in it.into_iter() {
+        let y = x.try_into();
+        if y.is_err() {
+            return Err(y.unwrap_err());
+        }
+        res.push(y.unwrap());
```

Comment:
```suggestion
        res.push(x.try_into()?);
```
I do not understand why clippy does not catch it, if we can easily remove unwrap we **should** remove unwrap

even more, everything can be mapped like this:
```rust
    it.into_iter()
        .map(|item| item.try_into())
        .collect::<Result<Vec<_>, _>>()
```

Reply:
Oh nice I was not aware we could collect an iterator of result into a result of vector.
I was looking for this on the rust Iterator documentation, there was try_collect which is not stable yet.
Nice to know.

@fanatid However since I checked before that is_err() was false before unwrap should it be the same as `try_into()?`?
If there is an error and we try unwrap/try_into()? it will panic either way.